### PR TITLE
EIO-247: security docs for 2.3.4

### PR DESCRIPTION
**Pull request by paraglade**

### CODE REVIEW

File: content/rancher/v2.x/en/security/hardening-2.3.4/_index.md [hunk 1]
```diff
@@ -0,0 +1,491 @@
...
+        data-dir: /var/lib/etcd
+      extra_binds:
+        - '/var/lib/etcd:/var/lib/etcd'
+      gid: 52034
+      retention: 72h
+      snapshot: false
+      uid: 52034
+    kube_api:
+      always_pull_images: false
+      extra_args:
```
Comment (cloudnautique) [inline content/rancher/v2.x/en/security/hardening-2.3.4/_index.md]: This needs to use the options built into RKE.
https://rancher.com/docs/rke/latest/en/config-options/audit-log/

File: content/rancher/v2.x/en/security/benchmark-2.3.5/_index.md [hunk 2]
```diff
@@ -27,9 +27,9 @@ For more detail about each audit, including rationales and remediations for fail
 
 Rancher and RKE install Kubernetes services via Docker containers. Configuration is defined by arguments passed to the container at the time of initialization, not via configuration files.
 
-Scoring the commands is different in Rancher Labs than in the CIS Benchmark. Where the commands differ from the original CIS benchmark, the commands specific to Rancher Labs are provided for testing.
+Scoring the commands is different in Rancher Labs than in the CIS Benchmark. Where the commands differ from the original CIS benchmark, the commands specific to Rancher Labs are provided for testing. Only **scored** tests will be covered in this guide.
```
Comment (cloudnautique) [inline content/rancher/v2.x/en/security/benchmark-2.3.5/_index.md]: I do not understand what is being stated in the first sentence of this section.  Is it that when commands differ from the CIS we will provide an alternate version to work with containers?

File: content/rancher/v2.x/en/security/benchmark-2.3.5/_index.md [hunk 3]
```diff
@@ -27,9 +27,9 @@ For more detail about each audit, including rationales and remediations for fail
...
 
-When performing the tests, you will need access to the Docker command line on the hosts of all three RKE roles. The commands also make use of the the `jq` command to provide human-readable formatting.
+When performing the tests, you will need access to the Docker command line on the hosts of all three RKE roles. The commands also make use of the the `jq` and `kubectl` (with valid config) commands to provide human-readable formatting.
```
Comment (cloudnautique) [inline content/rancher/v2.x/en/security/benchmark-2.3.5/_index.md]: These look like pre-requisites for running the audit commands?

File: content/rancher/v2.x/en/security/benchmark-2.3.5/_index.md [hunk 4]
```diff
@@ -173,14 +137,14 @@ ps -ef | grep etcd
...
-``` bash 
+``` bash
 chown etcd:etcd /var/lib/etcd
 ``` 
 
 **Audit:**
 
 ```
-ps -ef | grep etcd | grep -- --data-dir | sed 's%.*data-dir[= ]\([^ ]*\).*%\1%' | xargs stat -c %U:%G
+/mnt/kube-bench/test_helpers/1.1.12.sh etcd
```
Comment (cloudnautique) [inline content/rancher/v2.x/en/security/benchmark-2.3.5/_index.md]: It is ok we are referencing scripts, but how does the user get these?  Where are they going to be published?

File: content/rancher/v2.x/en/security/hardening-2.3.5/_index.md [hunk 5]
```diff
@@ -0,0 +1,424 @@
...
+communicate with each other and other network endpoints.
+
+Network Policies are namespace scoped. When a network policy is introduced to a given
+namespace, all traffic not allowed by the policy is denied. However, if there are no network
+policies in a namespace all traffic will be allowed into and out of the pods in that
+namespace.
+
+> todo: add information about network policies and provide default example here:
+
+
```
Comment (cloudnautique) [inline content/rancher/v2.x/en/security/hardening-2.3.5/_index.md]: Can we have a paragraph explaining how and when to use the reference implementations below?

File: content/rancher/v2.x/en/security/hardening-2.3.5/_index.md [hunk 6]
```diff
@@ -96,11 +96,43 @@ communicate with each other and other network endpoints.
...
 policies in a namespace all traffic will be allowed into and out of the pods in that
-namespace.
+namespace. To use network policies, you must be using a networking solution which supports `NetworkPolicy`.
+A CNI (container network interface) plugin can provide the needed `NetworkPolicy` resource.
+For this guide [canal](https://github.com/projectcalico/canal) will be used to provide
+the `NetworkPolicy` resource. Additional information about CNI providers can be found
+[here](https://rancher.com/blog/2019/2019-03-21-comparing-kubernetes-cni-providers-flannel-calico-canal-and-weave/)
+
+Once a CNI provider is enabled on a cluster that supports the `NetworkPolicy` resouce a default network policy
+can be applied. For reference purposes a **permissive** example is provide below. If you want to
```
Comment (cloudnautique) [inline content/rancher/v2.x/en/security/hardening-2.3.5/_index.md]: I think we should explicitly say not for production use.

File: content/rancher/v2.x/en/security/hardening-2.3.5/_index.md [hunk 7]
```diff
@@ -96,11 +96,43 @@ communicate with each other and other network endpoints.
 Network Policies are namespace scoped. When a network policy is introduced to a given
 namespace, all traffic not allowed by the policy is denied. However, if there are no network
 policies in a namespace all traffic will be allowed into and out of the pods in that
-namespace.
+namespace. To use network policies, you must be using a networking solution which supports `NetworkPolicy`.
+A CNI (container network interface) plugin can provide the needed `NetworkPolicy` resource.
+For this guide [canal](https://github.com/projectcalico/canal) will be used to provide
+the `NetworkPolicy` resource. Additional information about CNI providers can be found
```
Comment (cloudnautique) [inline content/rancher/v2.x/en/security/hardening-2.3.5/_index.md]: `NetworkPolicy` enforcement. The resource is a Kubernetes thing.

File: content/rancher/v2.x/en/security/hardening-2.3.5/_index.md [hunk 8]
```diff
@@ -132,26 +132,19 @@ for namespace in $(kubectl get namespaces -A -o json | jq -r '.items[].metadata.
   kubectl apply -f default-allow-all.yaml -n ${namespace}
 done
 ```
-Execute this scipt to apply the `default-allow-all.yaml` the **permissive** `NetworkPolicy` to all namespaces.
+Execute this script to apply the `default-allow-all.yaml` the **permissive** `NetworkPolicy` to all namespaces.
 
-### Reference Hardened RKE `config.yml` configuration 
+### Reference Hardened RKE `cluster.yml` configuration
+The reference `cluster.yml` provides the configuration needed to acheive a hardened install
```
Comment (cloudnautique) [inline content/rancher/v2.x/en/security/hardening-2.3.5/_index.md]: call out that this is the CLI

File: content/rancher/v2.x/en/security/hardening-2.3.5/_index.md (hunk 7 above)
Comment (paraglade) [reply content/rancher/v2.x/en/security/hardening-2.3.5/_index.md]: changes added

File: content/rancher/v2.x/en/security/hardening-2.3.5/_index.md (hunk 8 above)
Comment (paraglade) [reply content/rancher/v2.x/en/security/hardening-2.3.5/_index.md]: added reference to cli

File: content/rancher/v2.x/en/security/benchmark-2.3.5/_index.md (hunk 4 above)
Comment (paraglade) [reply content/rancher/v2.x/en/security/benchmark-2.3.5/_index.md]: test helper script contents added

**Comment by catherineluse** [comment]
I found a couple of typos in the hardening guide:

> **generenally** used in cloud infrastructure environments to allow for configuration **managment** of compute instances. 

Also I'd like to see Kubernetes API server used in paragraphs and `kube-apiserver` only used to refer to the container. Otherwise LGTM

File: content/rancher/v2.x/en/security/hardening-2.3.5/_index.md [hunk 9]
```diff
@@ -0,0 +1,705 @@
...
+
+### Overview
+
+This document provides prescriptive guidance for hardening a production installation of Rancher v2.3.5 with Kubernetes v1.15. It outlines the configurations required to address Kubernetes benchmark controls from the Center for Information Security (CIS).
+
+For more detail about evaluating a hardened cluster against the official CIS benchmark, refer to the [CIS Benchmark Rancher Self-Assessment Guide - Rancher v2.3.5]({{< baseurl >}}/rancher/v2.x/en/security/benchmark-2.3.5/).
+
+### Configure Kernel Runtime Parameters
+
+The folowing `sysctl` configuration is recommended for all nodes type in the cluster. Set the following parameters in `/etc/sysctl.d/90-kubelet.conf`:
```
Comment (Oats87) [inline content/rancher/v2.x/en/security/hardening-2.3.5/_index.md]: nit: following

File: content/rancher/v2.x/en/security/hardening-2.3.5/_index.md [hunk 10]
```diff
@@ -0,0 +1,705 @@
...
+      generate_serving_certificate: true
+    scheduler:
+      extra_args:
+        address: 127.0.0.1
+        profiling: 'false'
+  ssh_agent_auth: false
+windows_prefered_cluster: false
+```
+
+### Hardened Reference Ubuntu **cloud-config**:
```
Comment (Oats87) [inline content/rancher/v2.x/en/security/hardening-2.3.5/_index.md]: Does this work for all Ubuntu versions?

**Pull request merged**